metaflow/plugins/kfp/kfp.py

```diff
@@ -231,15 +231,23 @@ def create_workflow_yaml(self, output_path: str, output_format: str = "kfp") ->
         if output_format == "kfp":
             pass  # KFP pipeline yaml is created by default
         elif output_format == "argo-workflow":
-            workflow["spec"]["serviceAccountName"] = (
-                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
-            )
+            # Output of KFP compiler already has workflow["kind"] = "Workflow".
+
+            # Keep generateName - Argo Workflow is usually used in single run.
+
+            # Service account is added through webhooks.
+            workflow["spec"].pop("serviceAccountName", None)
+
         elif output_format == "argo-workflow-template":
-            workflow["spec"]["serviceAccountName"] = (
-                KUBERNETES_SERVICE_ACCOUNT or "default-editor"
-            )
             workflow["kind"] = "WorkflowTemplate"
-            workflow["status"] = None
+
+            # Use static name to make referencing easier.
+            workflow["metadata"]["name"] = (
+                workflow["metadata"].pop("generateName").rstrip("-")
```

Comment:
can you please take a look at `resolve_workflow_name()` in https://github.com/zillow/metaflow/pull/236/files#diff-d0dd4e31d56c308214f7dd8cdfe10ce53984b49506eacdb555aac0b0a52121e4R613 gotten from https://github.com/Netflix/metaflow/blob/master/metaflow/plugins/argo/argo_workflows_cli.py#L271

Reply:
Thanks for linking the PR! 

With KFP compiler we already made sure the generateName is a valid K8s string. The compiler calls [sanitize_k8s_name](https://github.com/zillow/pipelines/blob/feature/zg/sdk/python/kfp/compiler/_k8s_helper.py#L20) for us. Thus it's safe to just use `generateName` from KFP compiler with a minor change here.

Let me see how we can reduce merge conflicts here...

Reply:
Had some quick discussion today over slack. I will be keeping my approach here. 

`resolve_workflow_name` will be helpful when we fully move away from KFP compilers.